modules/apps/frontend-js/frontend-js-test/src/testFunctional/tests/SiteLevelWalkthrough.testcase

```diff
@@ -0,0 +1,61 @@
+@component-name = "portal-frontend-infrastructure"
+definition {
+
+	property osgi.modules.includes = "frontend-js-walkthrough-sample-web";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Walkthrough";
+	property testray.main.component.name = "User Interface";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		PortalInstances.tearDownCP();
+	}
+
+	@description = "LPS-159836."
+	@priority = "5"
+	test WillDisplayWhenDefinedForSite {
+		property portal.acceptance = "true";
+
+		var portalURL = PropsUtil.get("portal.url");
+
+		task ("Given there is a site-level JSON defined for current site") {
+			JSONGroup.addGroup(groupName = "Site Name");
+
+			ApplicationsMenu.gotoSite(site = "Site Name");
+
+			Walkthrough.enableWalkthrough(fileName = "walkthrough_configuration.json");
+
+			JSONLayout.addPublicLayout(
+				groupName = "Site Name",
+				layoutName = "Walkthrough Page 1");
+
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Site Name",
+				layoutName = "Walkthrough Page 1",
+				widgetName = "JS Walkthrough Sample");
+
+			Navigator.openSitePage(
+				pageName = "Walkthrough Page 1",
+				siteName = "Site Name");
+		}
+
+		task ("When instantiate the walkthrough") {
+			Walkthrough.enablePopoverMode();
+		}
```

Comment:
Nit picking, but I believe instantiating occurs when user navigates to the page. cc/ @susanchen3 please correct me if I'm wrong.

So I'd include the page navigation step in the When step and move the enablePopovermode to the Then step.

Reply:
Sorry I didn't see this sooner, but yes this sounds good to me too